src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -41,17 +42,18 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 		@IInlineChatService private readonly _inlineChatService: IInlineChatService,
 		@IInlineChatSessionService private readonly _inlineChatSessionService: IInlineChatSessionService,
 		@IConfigurationService private readonly _configurationService: IConfigurationService,
-		@IKeybindingService readonly _keybindingService: IKeybindingService,
+		@IKeybindingService private readonly _keybindingService: IKeybindingService,
 	) {
 		super();
-		const keybindings = _keybindingService.lookupKeybinding('inlineChat.start')?.getLabel();
+		this._inlineChatKeybinding = this._keybindingService.lookupKeybinding('inlineChat.start')?.getLabel() ?? undefined;
```

Comment:
I think it's nice to avoid repeating the code that gets the keybinding. I would only have it in `_updateKeybinding`.
I would:
- in the constructor create the decoration without `glyphMarginHoverMessage`
- call `_updateKeybinding` at the end of the constructor (I would rename it `_updateDecorationHover`)
- in the constructor listen to `onDidUpdateKeybindings` and call `_updateKeybinding`
- in `_updateKeybinding` get the keybinding and set glyphMarginHoverMessage. Method `_hoverMessage` can be inlined there. Don't call _onEnablementOrModelChanged(), no need to reset teh featire (will remove and add the decoration). It is very unlikely that the keybinding changes while the hover is showing, we can ignore that.
Instead of concatenating the string when there is a key binding, better have two separate strings:
`Start Inline Chat` and `Start Inline Chat [{0}]`. That gives the translators the chance to place the keybinding label the way they want.

Reply:
So the reason I call `_onEnablementOrModelChanged` inside of the method is because if the keyboard shortcuts and the document are open side by side, then if you change the keyboard shortcuts the hover message will not be updated when you focus back on the document. It will update on the next model change.

Reply:
ok, I see, didn't think of that, makes sense.

Reply:
Can you still fix the construction of the label?

Reply:
Oh, sorry, now I see it. LOCALIZED_START_INLINE_CHAT_STRING can now also be inlined.

Reply:
Oh yes, I was reusing this string because it is also used for the command `Start Inline Chat` in the command palette